    ---
uint64 rangeFront_srv
uint64 rangeBack_srv
uint64 rangeRight_srv
uint64 rangeLeft_srv
